Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -0,0 +1,248 @@
+            "get_spawn_point_info",
+            [this](const GetSpawnPointInfoRequest request, GetSpawnPointInfoResponse response)
+            {
+                GetSpawnPointInfo(request, response);
+            });
+
+        m_getSpawnPointsNamesService = ros2Node->create_service<gazebo_msgs::srv::GetWorldProperties>(
+            "get_spawn_points_names",
+            [this](const GetSpawnPointsNamesRequest request, GetSpawnPointsNamesResponse response)
+            {
+                GetSpawnPointsNames(request, response);
+            });
+    }
+
+    void ROS2SpawnerComponent::Deactivate()
+    {
+        m_getSpawnablesNamesService.reset();
+        m_spawnService.reset();
+        m_getSpawnPointInfoService.reset();
+        m_getSpawnPointsNamesService.reset();
+    }
+
+    void ROS2SpawnerComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ROS2SpawnerComponent, AZ::Component>()
+                ->Version(1)
+                ->Field("Spawnables", &ROS2SpawnerComponent::m_spawnables)
+                ->Field("Default spawn point", &ROS2SpawnerComponent::m_defaultSpawnPose);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ROS2SpawnerComponent>("ROS2 Spawner", "Spawner component")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "Manages spawning of robots in configurable locations")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->DataElement(AZ::Edit::UIHandlers::EntityId, &ROS2SpawnerComponent::m_spawnables, "Spawnables", "Spawnables")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::EntityId,
+                        &ROS2SpawnerComponent::m_defaultSpawnPose,
+                        "Default spawn pose",
+                        "Default spawn pose");
+            }
+        }
+    }
+
+    void ROS2SpawnerComponent::GetAvailableSpawnableNames(
+        const GetAvailableSpawnableNamesRequest request, GetAvailableSpawnableNamesResponse response)
+    {
+        for (const auto& spawnable : m_spawnables)
+        {
+            response->model_names.emplace_back(spawnable.first.c_str());
+        }
+    }
+
+    void ROS2SpawnerComponent::SpawnEntity(const SpawnEntityRequest request, SpawnEntityResponse response)
+    {
+        AZStd::string spawnable_name(request->name.c_str());
+        // xml parameter of the request is used here like a regular string and stores name of a spawn point
+        // todo: use xml format in this parameter
+        AZStd::string spawn_point_name(request->xml.c_str());
+
+        auto spawn_points = GetSpawnPoints();
```

Comment:
These variable names should be lowerCamelCase.
Also the "todo" should be made into a GHI.
```suggestion
        AZStd::string spawnableName(request->name.c_str());
        // xml parameter of the request is used here like a regular string and stores name of a spawn point
        // todo: use xml format in this parameter
        AZStd::string spawnPointName(request->xml.c_str());

        auto spawnPoints = GetSpawnPoints();
```